File: Box.js
class Box {
    constructor(xLeft, xRight, yUp, yDown, r) {
        let horizLength = xRight - xLeft;
        let vertLength = yDown - yUp;
        if (horizLength <= 0 || vertLength <= 0) {
            throw "Invalid box locations, check the parameters";
        }
        this.borders = {
            up: new HorizontalBorder((xLeft + xRight) / 2, yUp, horizLength, r),

            down: new HorizontalBorder((xLeft + xRight) / 2, yDown, horizLength, r),

            left: new VerticalBorder(xLeft, (yUp + yDown) / 2, vertLength, r),

            right: new VerticalBorder(xRight, (yUp + yDown) / 2, vertLength, r),
        };

        this.borders.up.extendTo(Way.up);
        this.borders.down.extendTo(Way.down);
        this.borders.left.extendTo(Way.left);
        this.borders.right.extendTo(Way.right);

    }

    draw() {
        for (let borderKey in this.borders) {
            this.borders[borderKey].draw();
        }
    }

    update() {}
}

File: pyrl/Box.py
from VerticalBorder import VerticalBorder
from HorizontalBorder import HorizontalBorder
from Way import Way

class Box:
    def __init__(self, x_left, x_right, y_up, y_down, restitution):
        horiz_length = x_right - x_left
        vert_length = y_down - y_up
        assert (horiz_length > 0 and vert_length > 0)
        
        self.borders = {
                "up": HorizontalBorder((x_left + x_right) / 2, y_up, horiz_length, restitution, visible=False),
                "down": HorizontalBorder((x_left + x_right) / 2, y_down, horiz_length, restitution, visible=False),
                "left": VerticalBorder(x_left, (y_up + y_down) / 2, vert_length, restitution, visible=False),
                "right": VerticalBorder(x_right, (y_up + y_down) / 2, vert_length, restitution, visible=False),
        }
        
        self.borders["up"].extend_to(Way.up)
        self.borders["down"].extend_to(Way.down)
        self.borders["left"].extend_to(Way.left)
        self.borders["right"].extend_to(Way.right)

    def draw(self):
        for border in self.borders.values():
            border.draw()
    
    def update(self):
        self